src/peergos/server/tests/simulation/NativeFileSystemImpl.java

```diff
@@ -70,7 +70,9 @@ private void ensureCan(Path path, Permission permission, String user) {
     public byte[] read(Path path, BiConsumer<Long, Long> pc) {
         Path nativePath = virtualToNative(path);
         ensureCan(path, Permission.READ);
-
+        if (nativePath.toFile().isDirectory()) {
+           return null;
```

Comment:
This would be easier to reason about if it raised an exception. Why return null?

Reply:
The method still confirms the file exists and has the right permissions. I didn't want to add read/write directory specific method to interface. But ok, message heard. I can do so.

Reply:
done now